Content.Server/Chat/ChatConditions/ActiveTelecomsEntityChatCondition.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Linq;
+using Content.Server.Power.Components;
+using Content.Server.Radio.Components;
+using Content.Server.Station.Components;
+using Content.Server.Station.Systems;
+using Content.Shared.Chat;
+using Content.Shared.Chat.Prototypes;
+using Content.Shared.Radio;
+using Content.Shared.Radio.Components;
+using Robust.Shared.Network;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Chat.ChatConditions;
+
+/// <summary>
+/// Checks if the consumer is on a map with an active server, if necessary.
+/// </summary>
+[DataDefinition]
+public sealed partial class ActiveTelecomsEntityChatCondition : EntityChatCondition
+{
+
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IEntityManager _entityManager = default!;
+
+    public override HashSet<EntityUid> FilterConsumers(HashSet<EntityUid> consumers, Dictionary<Enum, object> channelParameters)
+    {
+        IoCManager.InjectDependencies(this);
+        var returnConsumers = new HashSet<EntityUid>();
+
+        if (!channelParameters.TryGetValue(DefaultChannelParameters.SenderEntity, out var senderEntity) ||
+            !channelParameters.TryGetValue(DefaultChannelParameters.RadioChannel, out var radioChannel) ||
+            !_prototypeManager.TryIndex((string)radioChannel, out RadioChannelPrototype? radioPrototype) ||
+            !_entityManager.TryGetComponent<TransformComponent>((EntityUid)senderEntity, out var sourceTransform))
+            return new HashSet<EntityUid>();
+
+        var activeServer = radioPrototype.LongRange;
+
+        if (activeServer == false)
+        {
+            var servers = _entityManager.EntityQuery<TelecomServerComponent, EncryptionKeyHolderComponent, ApcPowerReceiverComponent, TransformComponent>();
+            var sourceMapId = sourceTransform.MapID;
+            foreach (var (_, keys, power, transform) in servers)
+            {
+                if (transform.MapID == sourceMapId &&
+                    power.Powered &&
+                    keys.Channels.Contains(radioPrototype.ID))
+                {
+                    activeServer = true;
```

Comment:
I'd advice to not reuse variables, this is majorly harder to understand as there is acutally 2 variables - there is marker 'isActiveServcerRequired' which is determined by radioPrototype.LongRange, and can be inlined maybe (into line 39) and there is that makrer, if there is active server. Also if you found that required active server is existing, you might want to stop iterating and return early.